connect/mirror/src/main/java/org/apache/kafka/connect/mirror/MirrorCheckpointConfig.java

```diff
@@ -166,6 +172,28 @@ Duration consumerPollTimeout() {
         return Duration.ofMillis(getLong(CONSUMER_POLL_TIMEOUT_MILLIS));
     }
 
+    public Map<String, String> validate() {
+        Map<String, String> invalidConfigs = new HashMap<>();
+
+        // No point to validate when connector is disabled.
+        if (!getBoolean(ENABLED)) {
+            return invalidConfigs;
+        }
+
+        if (!this.getBoolean(EMIT_CHECKPOINTS_ENABLED) && !this.getBoolean(SYNC_GROUP_OFFSETS_ENABLED)) {
+            Arrays.asList(SYNC_GROUP_OFFSETS_ENABLED, EMIT_CHECKPOINTS_ENABLED).forEach(configName -> {
+                invalidConfigs.putIfAbsent(configName, "MirrorCheckpointConnector can't run without both " + SYNC_GROUP_OFFSETS_ENABLED + ", " +
+                            EMIT_CHECKPOINTS_ENABLED + " set to false");
+            });
+        }
+        if ("false".equals(Optional.ofNullable(this.originals().get(EMIT_OFFSET_SYNCS_ENABLED)).orElse("true"))) {
+            invalidConfigs.put(EMIT_OFFSET_SYNCS_ENABLED, "MirrorCheckpointConnector can't run with " + EMIT_OFFSET_SYNCS_ENABLED + " set to false while, " +
+                    EMIT_CHECKPOINTS_ENABLED  + " and/or" + SYNC_GROUP_OFFSETS_ENABLED + " set to true");
```

Comment:
I think we can simplify this; there's no need to mention checkpointing or offset syncing since they're implicitly required by the previous validation:

```suggestion
            invalidConfigs.put(EMIT_OFFSET_SYNCS_ENABLED, "MirrorCheckpointConnector can't run without offset syncs");
```